packages/web/src/components/notification/Notifications/EntityLink.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { MouseEventHandler, useCallback } from 'react'
+
+import { push } from 'connected-react-router'
+import { useDispatch } from 'react-redux'
+
+import { Name } from 'common/models/Analytics'
+import { Collection } from 'common/models/Collection'
+import { Track } from 'common/models/Track'
+import { User } from 'common/models/User'
+import { toggleNotificationPanel } from 'common/store/notifications/actions'
+import { Entity } from 'common/store/notifications/types'
+import { useRecord, make } from 'store/analytics/actions'
+import { getEntityLink } from 'utils/route'
+
+import styles from './EntityLink.module.css'
+
+type EntityType = (Collection | Track) & { user: User }
```

Comment:
I have a feeling this may need to support Tips and Users in the future 😄

Reply:
yeah that's the hope :D